app/models/user.rb

```diff
@@ -1,2 +1,28 @@
 class User < ApplicationRecord
+  has_secure_password
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+
+  validates :name, presence: true,
+length: {maximum: Settings.defaults.user.MAX_NAME_LENGTH}
+  validates :email, presence: true, length:
+                    {maximum: Settings.defaults.user.MAX_EMAIL_LENGTH},
+                    format: {with: VALID_EMAIL_REGEX},
+                    uniqueness: {case_sensitive: false}
+  validate :birthday_must_be_within_last_100_years
+
+  before_save :downcase_email
+
+  private
+
+  def downcase_email
+    email.downcase!
+  end
+
+  def birthday_must_be_within_last_100_years
+    return if birthday.blank?
+
+    return unless birthday < 100.years.ago.to_date || birthday > Time.zone.today
+
+    errors.add(:birthday, "must be within the last 100 years")
```

Comment:
```suggestion
    errors.add(:birthday, :gt_100_years_old)
```
Dùng key và add i18n messages cho key này nha gt_100_years_old